src/components/explorer/ExplorerDatasetSearch.js

```diff
@@ -1,265 +1,123 @@
-                sorter: (a, b) => a.individual.id.localeCompare(b.individual.id),
-                sortOrder: sortColumnKey === "individual" && sortOrder,
-                defaultSortOrder: "ascend",
-            },
-            {
-                title: "Samples",
-                dataIndex: "biosamples",
-                render: samples => <>
-                    {samples.length} Sample{samples.length === 1 ? "" : "s"}{samples.length ? ": " : ""}
-                    {samples.join(", ")}
-                </>,
-                sorter: (a, b) => a.biosamples.length - b.biosamples.length,
-                sortOrder: sortColumnKey === "biosamples" && sortOrder,
-                sortDirections: ["descend", "ascend", "descend"],
-            },
-            {
-                title: "Experiments",
-                dataIndex: "experiments",
-                render: experiments => <>{experiments} Experiment{experiments === 1 ? "" : "s"}</>,
-                sorter: (a, b) => a.experiments - b.experiments,
-                sortOrder: sortColumnKey === "experiments" && sortOrder,
-                sortDirections: ["descend", "ascend", "descend"],
-            },
-        ];
-
-        if (!this.props.match.params.dataset) return null;  // TODO
-
-        const selectedDataset = this.props.datasetsByID[this.props.match.params.dataset];
+    const onTabChange = (newActiveKey) => {
+        setActiveKey(newActiveKey);
+        handleSetSelectedRows([]);
+    };
 
-        if (!selectedDataset) return null;  // TODO
+    const resetPageNumber = () => {
+        setCurrentPage(1);
+        dispatch(performSearchIfPossible(dataset));
+    };
 
-        const numResults = (this.props.searchResults || {searchFormattedResults: []}).searchFormattedResults.length;
+    if (!dataset) return null; // TODO
 
-        const isFetchingSearchResults = this.props.fetchingSearch || this.props.fetchingTextSearch;
+    const selectedDataset = datasetsByID[dataset];
 
-        const tableStyle = {
-            opacity: (isFetchingSearchResults ? 0.5 : 1),
-            pointerEvents: (isFetchingSearchResults ? "none" : "auto"),
-        };
+    if (!selectedDataset) return null; // TODO
 
-        // Calculate page numbers and range
-        const showingResults = numResults > 0
-            ? (this.state.currentPage * this.state.pageSize) - this.state.pageSize + 1
-            : 0;
+    const isFetchingSearchResults = fetchingSearch || fetchingTextSearch;
 
-        console.log("search results: ", this.props.searchResults);
+    const hasIndividuals = searchResults && searchResults.searchFormattedResults;
+    const hasExperiments =
+        searchResults &&
+        searchResults.searchFormattedResultsExperiment &&
+        searchResults.searchFormattedResultsExperiment.length > 0;
+    const hasBiosamples =
+        searchResults &&
+        searchResults.searchFormattedResultsBiosamples &&
+        searchResults.searchFormattedResultsBiosamples.length > 0;
```

Comment:
Here and above (line 78) we can simplify the evaluation by removing `> 0`, since 0 is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)

Reply:
Thanks for the suggestion, however here the code is ensuring there is at least one exp/biosample available before setting hasExp/Biosample to true. This avoid to display tables without data

Reply:
Sorry I was not clear. What I meant is that I think it can be written like so:
```javascript
const hasBiosamples =
        searchResults &&
        searchResults.searchFormattedResultsBiosamples &&
        searchResults.searchFormattedResultsBiosamples.length;
```
Which should still evaluate to `false` when there is no data.

An even safer check would be:
```javascript
const hasBiosamples =
        searchResults &&
        Array.isArray(searchResults.searchFormattedResultsBiosamples) &&
        searchResults.searchFormattedResultsBiosamples.length;
```
This way we make sure `searchResults.searchFormattedResultsBiosamples` is an array, and that its `length` property is truthy (above 0).

Very minor suggestion, the original should be fine, but in javascript I prefer to be on the safer side with types.

Reply:
I see!! Thank you very much for the clarification, you are right, the safer option is better!

Reply:
good call with the `hasNonEmptyArrayProperty`